Roles/Impostor/DoubleAgent.cs

```diff
@@ -0,0 +1,358 @@
+                pc.Notify(Utils.ColorString(Utils.GetRoleColor(pc.GetCustomRole()), GetString("DoubleAgentRoleChange") + RoleName));
+            }
+        }
+
+        if (CurrentBombedPlayers.Any(playerId => Utils.GetPlayerById(playerId) == null)) // If playerId is a null Player clear bomb.
+            ClearBomb();
+    }
+
+    // Active bomb timer update and check.
+    private void OnFixedUpdateOthers(PlayerControl player)
+    {
+        if (!CurrentBombedPlayers.Contains(player.PlayerId)) return;
+
+        if (!player.IsAlive()) // If Player is dead clear bomb.
+            ClearBomb();
+
+        if (BombIsActive && (GameStates.IsInTask && GameStates.IsInGame) && !(GameStates.IsMeeting && GameStates.IsExilling))
+        {
+            var OldCurrentBombedTime = (int)CurrentBombedTime;
+
+            CurrentBombedTime -= Time.deltaTime;
+
+            if (OldCurrentBombedTime > (int)CurrentBombedTime && CurrentBombedTime < (BombExplosionTimer.GetFloat() + 1))
+                SendRPC();
+
+            if (CurrentBombedTime < 1)
+                BoomBoom(player);
+        }
+    }
+
+    // Set timer on Double Agent for Non-Modded Clients.
+    public override void OnFixedUpdateLowLoad(PlayerControl pc)
+    {
+        if (BombIsActive)
+        {
+            if (!pc.IsModClient())
+            {
+                string Duration = Utils.ColorString(pc.GetRoleColor(), string.Format(GetString("DoubleAgent_BombExplodesIn"), (int)CurrentBombedTime));
+                if ((!NameNotifyManager.Notice.TryGetValue(pc.PlayerId, out var a) || a.Item1 != Duration) && Duration != string.Empty) pc.Notify(Duration, 1.1f);
+            }
+        }
+    }
+
+    // Players go bye bye ¯\_(ツ)_/¯
+    private void BoomBoom(PlayerControl player)
+    {
+        if (player.inVent) player.MyPhysics.RpcBootFromVent(player.GetPlayerVentId());
+
+        foreach (PlayerControl target in Main.AllAlivePlayerControls) // Get players in radius of bomb that are not in a vent.
+        {
+            if (CheckForPlayersInRadius(player, target) <= ExplosionRadius.GetFloat())
+            {
+                if (player.inVent) return;
+                Main.PlayerStates[target.PlayerId].deathReason = PlayerState.DeathReason.Bombed;
+                target.RpcMurderPlayer(target);
+            }
+        }
+
+        PlaySoundForAll("Boom");
+        ClearBomb();
+
+        _Player.Notify(Utils.ColorString(Utils.GetRoleColor(CustomRoles.DoubleAgent), GetString("DoubleAgent_BombExploded")));
+    }
+
+    private static float CheckForPlayersInRadius(PlayerControl player, PlayerControl target) => Vector2.Distance(player.GetCustomPosition(), target.GetCustomPosition());
+
+    // Play specific sound for all players.
+    private static void PlaySoundForAll(string Sound)
+    {
+        foreach (PlayerControl player in Main.AllPlayerControls)
+        {
+            player.RPCPlayCustomSound(Sound);
+        }
+    }

```

Comment:
It's already in the code
![image](https://github.com/user-attachments/assets/5185f8d0-0a4b-40d9-80fd-fa333ec82d14)